src/Tizen.NUI.WindowSystem/src/public/SoftkeyService.cs

```diff
@@ -0,0 +1,312 @@
+﻿/*
+ * Copyright(c) 2020 Samsung Electronics Co., Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+using System;
+using System.ComponentModel;
+
+namespace Tizen.NUI.WindowSystem.Shell
+{
+    /// <summary>
+    /// Class for the Tizen softkey service.
+    /// </summary>
+    /// This class is need to be hidden as inhouse API.
+    [EditorBrowsable(EditorBrowsableState.Never)]
+    public class SoftkeyService : IDisposable
+    {
+        private TizenShell _tzsh;
+        private IntPtr _softkeyService;
+        private int _tzshWin;
+        private bool disposed = false;
+        private bool isDisposeQueued = false;
+
+        private Interop.SoftkeyService.SoftkeyVisibleEventCallback _onVisibleChanged;
+        private event EventHandler<VisibleState> _visibleChanged;
+        /// <summary>
+        /// Emits the event when the visible state of the softkey service window is changed.
+        /// </summary>
+        /// <exception cref="ArgumentException">Thrown when failed of invalid argument.</exception>
+        /// <exception cref="Tizen.Applications.Exceptions.OutOfMemoryException">Thrown when the memory is not enough to allocate.</exception>
+        /// <since_tizen> 8 </since_tizen>
+        public event EventHandler<VisibleState> VisibleChanged
+        {
+            add
+            {
+                Tizen.Log.Error("JEON", $"add visiblechanged");
```

Comment:
`JEON` is not acceptable key for dlog. It should be `SoftkeyService` if necessary.

Reply:
I removed dlog. This is my fault.